Assets/ModScripts/Submodules/PianoDecryption.cs

```diff
@@ -0,0 +1,226 @@
+            Debug.LogFormat("[The Cruel Modkit #{0}] Decrypted word: {1}. Beginning encryption.", ModuleID, Encryption[0]);
+            Debug.LogFormat("[The Cruel Modkit #{0}] Atbash Cipher: {1} -> {2}", ModuleID, Encryption[0], Encryption[1]);
+            if (rot13.Any(x => serialLetters.Contains(x)))
+            {
+                encryptedWord = rot13;
+                Encryption.Add(encryptedWord);
+                Debug.LogFormat("[The Cruel Modkit #{0}] ROT13 Cipher: {1} -> {2}", ModuleID, Encryption[1], Encryption[2]);
+            }
+
+            int X = Module.GetComponent<KMBombInfo>().GetIndicators().Count() * Module.GetComponent<KMBombInfo>().GetSerialNumberNumbers().First() + 1;
+            encryptedWord = AffineTimesXCipher(X, encryptedWord);
+            Debug.LogFormat("[The Cruel Modkit #{0}] Affine *X Cipher, X = {1}: {2} -> {3}", ModuleID, X, Encryption.Last(), encryptedWord);
+            Encryption.Add(encryptedWord);
+
+            if (Module.GetComponent<KMBombInfo>().GetPortCount() <= Module.GetComponent<KMBombInfo>().GetPortPlateCount())
+            {
+                encryptedWord = Key3RailFenceCipher(encryptedWord);
+                Debug.LogFormat("[The Cruel Modkit #{0}] Rail Fence Cipher, Key = 3: {1} -> {2}", ModuleID, Encryption.Last(), encryptedWord);
+                Encryption.Add(encryptedWord);
+            }
+
+            bool forward = Module.GetComponent<KMBombInfo>().GetIndicators().Count() == 0;
+            int offset = forward ? Module.GetComponent<KMBombInfo>().GetSerialNumberNumbers().Last() : Module.GetComponent<KMBombInfo>().GetIndicators().Count();
+            encryptedWord = CaesarShift(encryptedWord, offset, forward);
+            Debug.LogFormat("[The Cruel Modkit #{0}] Caesar Shift {1} time{2} {3}: {4} -> {5}", ModuleID, offset,
+                offset != 1 ? "s" : "", forward ? "forwards" : "backwards", Encryption.Last(), encryptedWord);
+
+            Debug.LogFormat("[The Cruel Modkit #{0}] Final encrypted word: {1}.", ModuleID, encryptedWord);
+            break;
+        }
+        Info.Morse = encryptedWord;
+        Module.StopCoroutine(Module.MorseCodeAnimation);
+        IEnumerator MorseCodeAnimation = Module.PlayWord(Info.Morse);
+        Module.StartCoroutine(MorseCodeAnimation);
+    }
+
+    string AtbashCipher(string w)
+    {
+        string e = "";
+        for (int i = 0; i < w.Length; i++) e += alpha[25 - alpha.IndexOf(w[i])];
+        return e;
+    }
+
+    string Rot13Cipher(string w)
+    {
+        string e = "";
+        for (int i = 0; i < w.Length; i++) e += alpha[(alpha.IndexOf(w[i]) + 13) % 26];
+        return e;
+    }
+
+    string AffineTimesXCipher(int X, string w)
+    {
+        string e = "";
+        for (int i = 0; i < w.Length; i++) e += alpha[(alpha.IndexOf(w[i]) * X) % 26];
+        return e;
+    }
+
+    string Key3RailFenceCipher(string w)
+    {
```

Comment:
Since the key is always 3, you can hardcode it instead of using the actual cipher